app/main.py

```diff
@@ -12,51 +12,45 @@ async def run_sequence(*functions: Awaitable[Any]) -> None:
         await function
 
 
-async def run_parallel(*functions: Awaitable[Any]) -> None:
-    await asyncio.gather(*functions)
+async def run_parallel(*functions: Awaitable[Any]) -> Tuple[Any]:
+    return await asyncio.gather(*functions)
 
 
 async def main() -> None:
-    # create an IOT service
     service = IOTService()
 
-    # create and register a few devices
     hue_light = HueLightDevice()
     speaker = SmartSpeakerDevice()
     toilet = SmartToiletDevice()
 
-    hue_light_id, speaker_id, toilet_id = await asyncio.gather(
+    results = await run_parallel(
         service.register_device(hue_light),
         service.register_device(speaker),
         service.register_device(toilet)
     )
 
-    # create a few programs
-    async def wake_up_program() -> None:
-        await run_parallel(
-            service.send_msg(Message(hue_light_id, MessageType.SWITCH_ON)),
-            run_sequence(
-                service.send_msg(Message(speaker_id, MessageType.SWITCH_ON)),
-                service.send_msg(Message(
-                    speaker_id,
-                    MessageType.PLAY_SONG,
-                    "Rick Astley - Never Gonna Give You Up"
-                )),
-            ),
-        )
-
-    async def sleep_program() -> None:
-        await run_parallel(
-            service.send_msg(Message(hue_light_id, MessageType.SWITCH_OFF)),
-            service.send_msg(Message(speaker_id, MessageType.SWITCH_OFF)),
-            run_sequence(
-                service.send_msg(Message(toilet_id, MessageType.FLUSH)),
-                service.send_msg(Message(toilet_id, MessageType.CLEAN)),
-            ),
-        )
+    hue_light_id, speaker_id, toilet_id = results[0], results[1], results[2]
```

Comment:
you can directly do hue_light_id, speaker_id, toilet_id = await run_parallel... because gather returns a list of result and python will automatically split this list to separate variables ;)